src/gui/systray.cpp

```diff
@@ -164,6 +140,46 @@ void Systray::create()
     }
 }
 
+void Systray::setupContextMenu()
+{
+    const auto oldContextMenu = _contextMenu.data();
+    _contextMenu = new QMenu();
+
+    if (AccountManager::instance()->accounts().isEmpty()) {
+        _contextMenu->addAction(tr("Add account"), this, &Systray::openAccountWizard);
+    } else {
+        _contextMenu->addAction(tr("Open main dialog"), this, &Systray::openMainDialog);
+    }
+
+    auto pauseAction = _contextMenu->addAction(tr("Pause sync"), this, &Systray::slotPauseAllFolders);
+    auto resumeAction = _contextMenu->addAction(tr("Resume sync"), this, &Systray::slotUnpauseAllFolders);
+    _contextMenu->addAction(tr("Settings"), this, &Systray::openSettings);
+    _contextMenu->addAction(tr("Help"), this, &Systray::openHelp);
+    _contextMenu->addAction(tr("Exit %1").arg(Theme::instance()->appNameGUI()), this, &Systray::shutdown);
+    setContextMenu(_contextMenu);
+
+    connect(_contextMenu, &QMenu::aboutToShow, [=] {
+        const auto folders = FolderMan::instance()->map();
+
+        const auto allPaused = std::all_of(std::cbegin(folders), std::cend(folders), [](Folder *f) { return f->syncPaused(); });
+        const auto pauseText = folders.size() > 1 ? tr("Pause sync for all") : tr("Pause sync");
+        pauseAction->setText(pauseText);
+        pauseAction->setVisible(!allPaused);
+        pauseAction->setEnabled(!allPaused);
+
+        const auto anyPaused = std::any_of(std::cbegin(folders), std::cend(folders), [](Folder *f) { return f->syncPaused(); });
+        const auto resumeText = folders.size() > 1 ? tr("Resume sync for all") : tr("Resume sync");
+        resumeAction->setText(resumeText);
+        resumeAction->setVisible(anyPaused);
+        resumeAction->setEnabled(anyPaused);
+    });
+
+    // If we delete the old QMenu before setting the new one the client will crash on GNOME.
+    if(oldContextMenu) {
+        delete oldContextMenu;
```

Comment:
@claucambra Can you do it in the place where you assign to `oldContextMenu` by using `QObject::deleteLater()` https://doc.qt.io/qt-5/qobject.html#deleteLater? Should do the same thing but, without this extra code at the end.

Reply:
Works great! TIL :)